build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/datastore/DatastoreAdapter.java

```diff
@@ -72,7 +73,9 @@ public BuildConfigSetRecord saveBuildConfigSetRecord(BuildConfigSetRecord buildC
      * @return The latest revision of the given build configuration
      */
     public BuildConfigurationAudited getLatestBuildConfigurationAudited(Integer buildConfigurationId) {
-        return datastore.getLatestBuildConfigurationAudited(buildConfigurationId);
+        BuildConfigurationAudited buildConfigAudited = datastore.getLatestBuildConfigurationAudited(buildConfigurationId);
+        buildConfigAudited.getProject().getBuildConfigurations().forEach(BuildConfiguration::getId);
```

Comment:
@michalszynkiewicz @matejonnet  If we are going to keep the LAZY fetching, I saw an earlier implementation. It basically does not get rid of the loop, but it's inside a dedicated class:  pnc/rest/src/main/java/org/jboss/pnc/rest/utils/HibernateLazyInitializer.java used inside  pnc/rest/src/main/java/org/jboss/pnc/rest/trigger/BuildTriggerer.java